### ISSUE TITLE
Fails to start after 2.3.0.2 (45183) update

### ISSUE THREAD

**Issue by sdcote** [docker/for-win#6698]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: D3C90128-6452-499D-8024-DBBA94D42A36/20200513114905

### Actual behavior
Switch delete failed, switch = '8D5B0BB4-E903-4A57-8755-518388B8626C': General access denied error (0x80070005).
<img width="514" alt="2020-05-13 16_31_24-Error" src="https://user-images.githubusercontent.com/564533/81862147-41eccf80-9537-11ea-8550-0f8e67d6b6b4.png">

See Attached:
[20200513114905.zip](https://github.com/docker/for-win/files/4624461/20200513114905.zip)

### Information
This happens every time Windows starts.
Tried restarting from the systray - same errors.
Tried to reset to defaults - same errors.
Tried to uninstall - hangs indefinitely trying to remove VM and images
Tried to reinstall - "Existing application is up to date"
Rebooted a few times, both on and off the network, trying all the above. - No joy.

I appear to have bricked Docker with this update. I can't run it nor remove it from my system.

  - Windows Version: Windows 10 Enterprise 1809 (OS Build17763.1158)
  - Docker Desktop Version: 2.3.0.2 (45183) 
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No

### Steps to reproduce the behavior

  1. Start Windows
  2. wait for error dialog

**Comment by Richard-Barrett** [comment]
# Windows
Here is what worked for me:
If you want to do it on Windows I would do the following:
1. Uninstall Everything Related to Docker-Desktop in Control Panel and disable Hyper-V
https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v
2. If you are using Chocolatey you can uninstall all of that there too with `choco uninstall package` and disable Hyper-V https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v
3. Restart your Machine and Enter BIOS Menu and Make sure you have Virtualization and/or Hyperthreading Enabled. 

Use this link to see how to get into BIOS: https://support.lenovo.com/us/en/solutions/ht500006
You may need to reset your BIOS to default (Note not all Machines Support Virtualization) 
4. Allow for normal boot to occur
5. Enable Hyper-V for Windows 10 
https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v
6. Run through the following in an elevated command prompt and allowing for administrative access by right-clicking on command prompt:
```
dism.exe /Online /Enable-Feature:Microsoft-Hyper-V /All
bcdedit /set hypervisorlaunchtype auto
```

7. Restart Your Computer One More Time
8. Boot up
9. You can go to Docker-Desktop or you can install it with Chocolatey
If you want to install it via website, I assume you know where to go. Get the Stable Version. 
10 (Option A). Install as normal via https://www.docker.com/products/docker-desktop
10 (Option B). Use Chocolatey: 
Open up powershell in elevated admin console and install chocolatey
```
Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
```
https://chocolatey.org/install

Then after you install it run:
```
choco install docker-desktop
```
Chocolatey Link: https://chocolatey.org/packages/docker-desktop#releasenotes
Docker Release Notes: https://docs.docker.com/docker-for-windows/release-notes/

Whatever you decide, you need to make sure 
1. Your System Supports Virtualization
2. You have that enabled in System BIOS
3. Makes sure you have downloaded the WSL 2 background: https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel

Hope this helps, I too have had so many problems with Docker-Desktop as of late.

**Comment by sdcote** [comment]
I cannot execute step #1, it hangs:

<img width="530" alt="2020-05-14 11_28_35-Uninstalling Docker Desktop 2 3 0 2 (45183)" src="https://user-images.githubusercontent.com/564533/81953835-2553a480-95d6-11ea-824e-f7c6a2bc539d.png">

**Comment by Richard-Barrett** [comment]
If uninstall fails, there is a community forum here for that: 
https://forums.docker.com/t/cannot-uninstall-docker-for-windows/15661/10

**Comment by Richard-Barrett** [comment]
You might want to check your machine and see if all the directories match what you have before you run it.

**Comment by sdcote** [comment]
This did not resolve the issue.  I still cannot start Docker. It still fails with "Failed while removing virtual Ethernet switch." due to a "General access denied error (0x80070005)."

**Comment by Richard-Barrett** [comment]
Interesting, you might try checking for a stale VM switch, read through the following it might help in your instance:

- https://superuser.com/questions/1192492/how-to-remove-a-hyper-v-virtual-ethernet-switch

- https://forums.docker.com/t/hyper-v-was-unable-to-find-a-virtual-switch-with-name-dockernat/42615/6

- https://social.technet.microsoft.com/Forums/en-US/aaa81680-809d-4c8d-a186-3f319cae399a/virtual-switch-manager-anomaly-failed-while-adding-virtual-ethernet-switch-connections?forum=virtualmachingmgrhyperv

**Comment by sdcote** [comment]
I did finally get the application to uninstall and reinstalled Docker.  I tried several things, but it seems that there was a registry entry which did not match my list and I deleted it.  The following article helped the most:

* https://superuser.com/questions/1192492/how-to-remove-a-hyper-v-virtual-ethernet-switch

The registry entry containing the virtual ethernet switch was located in:
```
HKLM\SYSTEM\CurrentControlSet\Services\VMSMP\Parameters\SwitchList
```
I'm going to close this Issue since it does not appear to be a common issue, but a one-off event on my workstation.
